vertx-web/src/main/java/io/vertx/ext/web/handler/sockjs/impl/BaseTransport.java

```diff
@@ -156,7 +156,11 @@ static void setJSESSIONID(SockJSHandlerOptions options, RoutingContext rc) {
     }
   }
 
-  static void setCORS(RoutingContext rc) {
+  static void setCORS(SockJSHandlerOptions options, RoutingContext rc) {
+    if (!options.isCorsHeadersEnabled()) {
```

Comment:
The intention of this change is not to remove duplicate headers. Instead, it is to completely remove `Access-Control-*` headers from the responses. This is helpful requests / connections must only ever be allowed from a single origin and when that origin is the origin of the WS socket.

This effectively makes authentication / security handlers easier as those wouldn't be required to check `Origin` headers (the browser would already enforce this security model).